File: app/src/at-myhuynh/java/asiantech/internship/summer/ex05/ListPrime.java
package asiantech.internship.summer.ex05;

import asiantech.internship.summer.common.Common;

public class ListPrime {

	public static void main(String[] args) {
		int n = Common.input("Nhập vào số n: ");
		print(n);
	}

	public static void print(int number) {
		for (int i = 1; i <= number; i++) {
			if (Common.isPrime(i)) {
				System.out.println(i);
			}
		}
	}
}

File: app/src/at-myhuynh/java/asiantech/internship/summer/kotlin/ex24/ListPrime.kt
package asiantech.internship.summer.kotlin.ex24

import asiantech.internship.summer.kotlin.common.isChainOfSymmetry
import asiantech.internship.summer.kotlin.common.isPrime

object ListPrime {

    @JvmStatic
    fun main(args: Array<String>) {
        (10000..9999999)
                .filter { it.isPrime() && it.toString().isChainOfSymmetry() && checkItemIsPrime(it) }
                .forEach(::println)
    }

    private fun checkItemIsPrime(n: Int): Boolean {
        val arrNum = n.toString().split("\\s+".toRegex())
        for (i in arrNum.indices) {
            if (!arrNum[i].toInt().isPrime()) {
                return false
            }
        }
        return true
    }
}
